sw/device/tests/dif/dif_gpio_test.cc

```diff
@@ -0,0 +1,420 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mock_mmio.h"
+
+extern "C" {
+#include "gpio_regs.h"  // Generated
+#include "sw/device/lib/dif/dif_gpio.h"
+}  // extern "C"
+
+// Base class for the test fixtures in this file.
+class DifGpioTest : public testing::Test, public mock_mmio::MmioTest {};
+
+// Init tests
+class InitTest : public DifGpioTest {};
+
+TEST_F(InitTest, NullArgs) {
+  dif_gpio_config_t config{.base_addr = dev().region()};
```

Comment:
I think this would make more sense if `dif_gpio_init` had a `base_addr` parameter. Since `base_addr` is part of the `config` struct, I will keep this part as is. This will need to be updated anyways if we take `base_addr` out of `config`.